diff --git a/src/StreamWrapper/S3fsStream.php b/src/StreamWrapper/S3fsStream.php
index 7a4a98c..9c35d8b 100644
--- a/src/StreamWrapper/S3fsStream.php
+++ b/src/StreamWrapper/S3fsStream.php
@@ -587,7 +587,8 @@ class S3fsStream extends StreamWrapper implements StreamWrapperInterface {
     }
     else {
       // No special request given, we can generate the link.
-      if (empty($this->config['use_cname'])) {
+      //if (empty($this->config['use_cname'])) {
+      if (empty($this->config['use_cname']) || strpos($s3_key,".mp4") == false) {
         try {
           $external_url = $this->s3->getObjectUrl($this->config['bucket'], $s3_key);
         }
