diff --git a/src/StreamWrapper/S3fsStream.php b/src/StreamWrapper/S3fsStream.php
index 40649af..8e049b4 100644
--- a/src/StreamWrapper/S3fsStream.php
+++ b/src/StreamWrapper/S3fsStream.php
@@ -535,7 +535,8 @@ class S3fsStream extends StreamWrapper implements StreamWrapperInterface {
     }
     else {
       // No special request given, we can generate the link.
-      if (empty($this->config['use_cname'])) {
+      //if (empty($this->config['use_cname'])) {
+      if (empty($this->config['use_cname']) || strpos($s3_key,".mp4") == false) {
         $external_url = $this->s3->getObjectUrl($this->config['bucket'], $s3_key);
         if (empty($this->config['use_https'])) {
           // Forced HTTPS not enabled and not an HTTPS page load.
